test/automate/automate-utils.js

```diff
@@ -0,0 +1,233 @@
+    .loggingTo('./chromedrive.log')
+    .enableVerboseLogging()
+    .build();
+
+  let options = new chrome.Options();
+  // configure browser options ...
+
+  let driver = chrome.Driver.createSession(options, service);
+
+  // Create the browser, load the page, and give it some time to complete.
+  /*
+  var driver = new webdriver.Builder()
+    .forBrowser('chrome')
+    .setChromeOptions()
+    .build();
+  */
+
+  driver.get(URL);
+  driver.sleep(5000);
+
+  // This next section removes bundle.js, and then injects our local version
+  // of bundle.js.
+
+  // Click on the "Edit Settings" button.
+  driver.wait(
+    until.elementLocated(By.id('edit-settings')),
+    elementTimeout
+  ).click();
+
+  // Wait for the item settings modal.
+  driver.wait(
+    until.elementLocated(By.id('item-settings-modal')),
+    elementTimeout
+  );
+
+  // Click on "Behavior" settings tab.
+  driver.wait(
+    until.elementLocated(By.id('settings-behavior-tab')),
+    elementTimeout
+  ).click();
+
+  // Make sure "Auto-Updating Preview" is checked. With it checked, we do not
+  // need to click the "Run" button after making changes.
+  var elementAutoRun = driver.wait(
+    until.elementLocated(By.id('auto-run')),
+    elementTimeout
+  );
+
+  elementAutoRun.getAttribute('checked')
+  .then(function(checked) {
+    if (!checked) {
+      elementAutoRun.click();
+    }
+  });
+
+  driver.sleep(5000);
+
+  // Click on javascript settings.
+  driver.wait(
+    until.elementLocated(By.id('settings-js-tab')),
+    elementTimeout
+  ).click();
+
+  // Find the bundle.js input row and set it to blank.
+  // TODO: Put the var declaration elsewhere.
+  var javascriptRows = driver.findElements(
+    By.className('js-resource external-resource tt-input')
+  );
+
+  javascriptRows.then(function(webElems) {
+    webElems.forEach(function(elem) {
+      var value = elem.getAttribute('value');
+      value.then(function(val) {
+        if (val.includes('bundle.js', 0)) {
+          elem.clear();
+          elem.sendKeys('https://127.0.0.1:8080/bundle.js');
+        }
+      });
+    });
+  });
+
+  driver.sleep(5000);
+
+  // Close the modal.
+  driver.wait(
+    until.elementLocated(By.id('close-settings')),
+    elementTimeout
+  ).click();
+
+  driver.sleep(5000);
+
+  // Re-run the web page.
+  /*
+  driver.wait(
+    until.elementLocated(By.id('run')),
+    elementTimeout
+  ).click();
+  */
+
+  // Inject the local version of bundle.js.
+  /*
+  var bundleScript = fs.readFileSync(
+    './build/bundle.js',
+    'utf8'
+  );
+
+  driver.sleep(5000);
+
+  driver.executeScript(bundleScript)
+  .then(function(returnVal) {
+    console.log('executeScript returnVal');
+    console.log(returnVal);
+  });
+
+  driver.sleep(10000);
+  */
+
+  // Change the CodePen view to full view so it is easier to see errors. And
+  // give the page a couple seconds to adjust the layout.
+  /*
+  driver.wait(
+    until.elementLocated(By.id('view-switcher-button')),
```

Comment:
I was going to say, the screenshots were cut off due to editor view, but here it is commented out. :)